blockmanager.go

```diff
@@ -1451,8 +1688,12 @@ out:
 				}
 
 			case processBlockMsg:
-				forkLen, isOrphan, err := b.cfg.Chain.ProcessBlock(
-					msg.block, msg.flags)
+				var isOrphan bool
+				forkLen, err := b.cfg.Chain.ProcessBlock(msg.block, msg.flags)
+				if blockchain.IsErrorCode(err, blockchain.ErrMissingParent) {
+					isOrphan = true
+					err = nil
+				}
```

Comment:
Was testing this out in simnet and I believe you need to `processOrphanBlocks`

Reply:
True.  I was considering doing that and the primary reason I decided not to was that orphans are rejected anyway if they are submitted via `getwork` or the CPU miner, but it's a good point that it's useful to retain for `submitblock`.  I'll get it updated here in a bit.

Reply:
Update to refactor the process block and orphan handling logic in block manager to a separate function that both places call.

Reply:
Output of testing the scenario in question with simnet:
```
2019-11-22 12:47:42.167 [INF] BMGR: Adding orphan block 02e1e08259ca56d43409cb1fab89861813d3fe7c1727302fc20e507ac6df8bf8 with parent 01389f17afc276b3c965c6f11c710879b20e17ea90e83d1927acab1bec6ab644
2019-11-22 12:47:42.167 [INF] RPCS: Accepted block 02e1e08259ca56d43409cb1fab89861813d3fe7c1727302fc20e507ac6df8bf8 via submitblock
2019-11-22 12:48:06.589 [INF] RPCS: Accepted block 01389f17afc276b3c965c6f11c710879b20e17ea90e83d1927acab1bec6ab644 via submitblock
```
```
$ ./ctl getbestblock
{
  "hash": "02e1e08259ca56d43409cb1fab89861813d3fe7c1727302fc20e507ac6df8bf8",
  "height": 34
}
```